src/pz390.java

```diff
@@ -4229,50 +4349,136 @@ private void ins_B2XX(){
-			break;
+            psw_check = false;
+            ins_setup_s();
+            if (psw_problem_state == psw_problem_mode)
+               {set_psw_check(psw_pic_priv); // RPI 1622 DK Privileged operation exception
+                } // in problem state issue privileged-operation exception, otherwise no-op as we have no TLB implemented
+            else // supervisor mode applies
+               {psw_check = true; // force S0C1 until we have a proper implementation
+                }
+            break;
 		case 0x10: // 2670 "B210" "SPX" "S"
-			ins_setup_s();
-			break;
+            psw_check = false;
+            ins_setup_s();
+            if (psw_problem_state == psw_problem_mode)
+               {set_psw_check(psw_pic_priv); // RPI 1622 DK Privileged operation exception
+                } // in problem state issue privileged-operation exception, otherwise no-op as we have no real/absolute storage distinction
+            else // supervisor mode applies
+               {psw_check = true; // force S0C1 until we have a proper implementation
+                }
+            break;
 		case 0x11: // 2680 "B211" "STPX" "S"
-			ins_setup_s();
-			break;
+            psw_check = false;
+            ins_setup_s();
+            if (psw_problem_state == psw_problem_mode)
+               {set_psw_check(psw_pic_priv); // RPI 1622 DK Privileged operation exception
+                } // in problem state issue privileged-operation exception, otherwise no-op as we have no support for real vs. absolute storage
+            else // supervisor mode applies
+               {psw_check = true; // force S0C1 until we have a proper implementation
+                }
+            break;
 		case 0x12: // 2690 "B212" "STAP" "S"
-			ins_setup_s();
-			break;
+            psw_check = false;
+            ins_setup_s();
+            if (psw_problem_state == psw_problem_mode)
+               {set_psw_check(psw_pic_priv); // RPI 1622 DK Privileged operation exception
+                } // in problem state issue privileged-operation exception, otherwise no-op as we have no CPU address
+            else // supervisor mode applies
+               {psw_check = true; // force S0C1 until we have a proper implementation
+                }
+            break;
 		case 0x18: // 2700 "B218" "PC" "S"
 			psw_check = false;
 			ins_setup_s();
 			push_pc_stack(true,psw_loc);
 			set_psw_loc(bd2_loc);
 			break;
 		case 0x19: // 2710 "B219" "SAC" "S"
-			ins_setup_s();
-			break;
+            psw_check = false;
+            ins_setup_s();
+            if (psw_problem_state == psw_problem_mode)
+               {set_psw_check(psw_pic_priv); // RPI 1622 DK Privileged operation exception
+                } // in problem state issue privileged-operation exception, otherwise no-op as we have no address spaces
+            else // supervisor mode applies
+               {psw_check = true; // force S0C1 until we have a proper implementation
+                }
+            break;
```

Comment:
SAC is used by any program to set Access Register Mode. The privileged operation only applies for an attempt to set the home=space mode in the problem state.

Reply:
That is completely correct. However, z390 does supports neither virtual memory nor even address spaces. The concept of AR mode, home space mode, etc. are undefined in the z390 context. Therefore any attempt to use SAC should result in an abend to warn the programmer that the program cannot execute as intended.

Reply:
No argument about issuing an exception. However, it is my opinion that the standard z390 technique of S0C1 via
```
case <instruction>:
    ins_setup_xxx();
    break;
```
is more appropriate for SAC. I'll defer to your judgment on how to do it.